### DISCUSSION TITLE
Seeking Assistance with Custom UserObject Execution between InversePowerMethod Iterations

### DISCUSSION THREAD

**Discussion by hityyds** [idaholab/moose#26933] [Q&A General]
Hello everyone,

I am currently working with the InversePowerMethod executioner in MOOSE and have encountered an issue regarding the execution timing of a custom userobject. I want it to be executed between power iterations. When using the InversePowerMethod, the iteration process looks like this:
```
Power iteration= 0
 0 Nonlinear |R| = 1.478162e-02
      0 Linear |R| = 1.478162e-02
      1 Linear |R| = 1.755819e-07
 1 Nonlinear |R| = 1.755819e-07

 +================+=====================+=====================+
 | iteration      | eigenvalue          | solution_difference |
 +================+=====================+=====================+
 |              0 |      4.29166712e-01 |      1.86513876e-03 |
 +================+=====================+=====================+
 Chebyshev step: 0
 ________________________________________________________________________________ 
 Power iteration= 1
 0 Nonlinear |R| = 1.776508e-02
      0 Linear |R| = 1.776508e-02
      1 Linear |R| = 9.905756e-08
 1 Nonlinear |R| = 9.905756e-08
```
Specifically, I am wanting my userobject to be executed before the '0 Nonlinear |R|' of each power iteration.
I've tried setting execute_on to NONLINEAR, but it only causes my userobject to be executed before the Linear iteration. The [SetupInterface](https://mooseframework.inl.gov/source/interfaces/SetupInterface.html#) page describes common Execute Flags, but none of them are what I need. I know there are other Execute Flags such as FORWARD, ADJOINT, HOMOGENEOUS_FORWARD, ADJOINT_TIMESTEP_BEGIN, ADJOINT_TIMESTEP_END, etc. How do I find out what these Execute Flags do? 

Additionally, considering that none of the standard flags fulfill my need, I suspect I might need to define a custom Execute Flag. However, the information on creating custom Execute Flags in the [SetupInterface](https://mooseframework.inl.gov/source/interfaces/SetupInterface.html#creating-custom-execute-flags) is quite brief. Can anyone recommend a more detailed tutorial or resource on how to create and utilize a custom Execute Flag in this context?

Your help and insights would be greatly appreciated!

**Comment by GiudGiud** [comment]
Hello

> However, the information on creating custom Execute Flags in the [SetupInterface](https://mooseframework.inl.gov/source/interfaces/SetupInterface.html#creating-custom-execute-flags) is quite brief.

but that s nice because this is litterally all you need to do to get it work. We cant get more detailed, you do those 3 steps and you can use it in your input already

Looking through the repo, LevelSet and XFEM modules have examples of doing this (use grep to find them)
also Cardinal does it too 
https://github.com/neams-th-coe/cardinal
```
src/base/CardinalAppTypes.C:const ExecFlagType EXEC_SEND_OPENMC_DENSITIES = registerExecFlag("SEND_OPENMC_DENSITIES");
```

Guillaume
